web/src/user_status.ts

```diff
@@ -0,0 +1,144 @@
+import {z} from "zod";
+
+import * as channel from "./channel";
+import * as emoji from "./emoji";
+import type {EmojiRenderingDetails} from "./emoji";
+import {user_settings} from "./user_settings";
+
+export type UserStatus = z.infer<typeof user_status_schema>;
+export type UserStatusEmojiInfo = EmojiRenderingDetails & {
+    emoji_alt_code?: boolean;
+};
+
+type UserStatusEvent = {
+    id: number;
+    type: "user_status";
+    user_id: number;
+    away?: boolean;
+    status_text?: string;
+} & (
+    | {
+          emoji_name: string;
+          emoji_code: string | number;
+          reaction_type: string;
+      }
+    | {
+          emoji_name: undefined;
```

Comment:
What's the story with `emoji_name` being present but undefined? That doesn't seem to match the API documentation.

Reply:
Yeah my bad, I confused the event with actual user_status object, fixed this.